pslogin/src/main/scala/WorldSessionActor.scala

```diff
@@ -4329,55 +4332,59 @@ class WorldSessionActor extends Actor
+            deadState = DeadState.Release //cancel movement updates
+            PlayerActionsToCancel()
+            continent.GUID(player.VehicleSeated) match {
+              case Some(vehicle : Vehicle) if vehicle.MountedIn.isEmpty =>
+                vehicle.PassengerInSeat(player) match {
+                  case Some(0) =>
+                    vehicle.Position = pos
+                    LoadZonePhysicalSpawnPoint(zone, pos, Vector3.Zero, 0)
+                  case _ => //not seated as the driver, in which case we can't move
+                    deadState = DeadState.Alive
+                }
+              case None =>
+                player.Position = pos
+                //continent.AvatarEvents ! AvatarServiceMessage(continent.Id, AvatarAction.ObjectDelete(player.GUID, player.GUID))
+                LoadZonePhysicalSpawnPoint(zone, pos, Vector3.Zero, 0)
+              case _ => //seated in something that is not a vehicle or the vehicle is cargo, in which case we can't move
+                deadState = DeadState.Alive
+            }
           }
-
         case (_, _, _) => ;
       }
+
       CSRWarp.read(traveler, msg) match {
-        case (true, pos) if player.isAlive =>
-          deadState = DeadState.Release //cancel movement updates
-          PlayerActionsToCancel()
-          continent.GUID(player.VehicleSeated) match {
-            case Some(vehicle : Vehicle) if vehicle.MountedIn.isEmpty =>
-              vehicle.PassengerInSeat(player) match {
-                case Some(0) =>
-                  vehicle.Position = pos
-                  CancelAllProximityUnits()
-                  LoadZonePhysicalSpawnPoint(continent.Id, pos, Vector3.z(vehicle.Orientation.z), 0)
-                case _ => //not seated as the driver, in which case we can't move
-                  deadState = DeadState.Alive
-              }
-            case None =>
-              player.Position = pos
-              CancelAllProximityUnits()
-              CancelZoningProcessWithDescriptiveReason("cancel_motion")
-              sendResponse(PlayerStateShiftMessage(ShiftState(0, pos, player.Orientation.z, None)))
-              deadState = DeadState.Alive //must be set here
-            case _ => //seated in something that is not a vehicle or the vehicle is cargo, in which case we can't move
-              deadState = DeadState.Alive
+        case (true, pos) =>
+          // continent.Id == "c1" || continent.Id == "c2" || continent.Id == "c3" || continent.Id == "c4" || continent.Id == "c5" || continent.Id == "c6" ||
+          if (player.isAlive && (admin || continent.Id == "z8" ||
+            continent.Id == "tzshtr" || continent.Id == "tzcotr" || continent.Id == "tzdrtr" ||
+            continent.Id == "tzshnc" || continent.Id == "tzconc" || continent.Id == "tzdrnc" ||
+            continent.Id == "tzshvs" || continent.Id == "tzcovs" || continent.Id == "tzdrvs")) {
+            deadState = DeadState.Release //cancel movement updates
```

Comment:
What's all this, then?

Up there with CSRZone too.

Reply:
Ported directly from Soun's PTSv3 branch. I think it's to allow non-admins to warp to certain zones like vr training or old oshur.